Natural language processing of time.  Add --timeframe, dev logging, d… (#49)
logdna/logdna-cli: README.md

# Search
$ logdna search "logdna cli" -a logdna.log -t tag1,tag2 -n 300
# Search via UNIX timestamps
$ logdna search "logdna" --from 1541100040931 --to 1541102940000
# Search via Natural Language (Chrono)

> # Ensure to enclose nl strings in ""

Reviewer: ```suggestion
# Ensure to enclose natural langauge strings in ""
```

Nit: People may not immediately get the acronym; generally, docs shouldn't have acronyms without explaining them at least once.

> # Must include date reference, IE May 26, yesterday or the like

Reviewer: ```suggestion
# Must include date reference (i.e., `May 26`, `yesterday`, or the like)
```

> $ logdna search "logdna" --from "yesterday at 3pm" to "today at 3:05pm"

Reviewer 2: This should be

```suggestion
$ logdna search "logdna" --from "yesterday at 3pm" --to "today at 3:05pm"
```

Author: fixed

> $ logdna search "logdna" --from "yesterday at 3pm" to "today at 3:05pm"

Reviewer 2: This should be

```suggestion
$ logdna search "logdna" --from "yesterday at 3pm" --to "today at 3:05pm"
```

Author: fixed

> Choose account [1-2]:
```

## Developer Logs

> Logs for troubleshooting during development are included.  These will pipe to both console and [LogDNA](https://www.logdna.com/).

Reviewer: ```suggestion
Logs for troubleshooting during development are included. These logs will pipe both to console and [LogDNA](https://www.logdna.com/).
```

Author: fixed

> Switch dev logs on via the `-d` or `--dev` flag

Reviewer: ```suggestion
Switch dev logs on via the `-d` or `--dev` flag.
```

Author: fixed

> Requires LogDNA account with environment variable LOGDNA_API_KEY see logs in LogDNA

Author: Fixed

> Specify a log as "dev" by using logger.log(...)

Reviewer: ```suggestion
Specify a log as "dev" by using `logger.log(...)`.
```

Author: fixed

> Note that we have used a standard log primitive via lib/utils.js -> devLogPrim() with "what", "why", "where", "who" and "when" to ensure full context is included as well as a "message" field for quick scanning in LogDNA.  See [logger initialization](index.js#L28), [error example](index.js#91) and [devLogPrim](lib/utils.js#L232) for more details.

Reviewer: Agreed. This could be an interesting blog post or Discourse discussion, though.

> Note that we have used a standard log primitive via lib/utils.js -> devLogPrim() with "what", "why", "where", "who" and "when" to ensure full context is included as well as a "message" field for quick scanning in LogDNA.  See [logger initialization](index.js#L28), [error example](index.js#91) and [devLogPrim](lib/utils.js#L232) for more details.

Author: moved to be a comment in index.js.  Love the idea of a Discourse discussion on the topic

> Currently the coverage is limited to index.js and more heavily in --timeframe, --from and --to.

Reviewer: ```suggestion
Currently the coverage is limited to index.js and more heavily in `--timeframe`, `--from`, and `--to`.
```

To be honest, I have no idea what this sentence means, but it may be that I just don't know the package (which is also true and very likely). What do you mean that it's limited to a specific file but heavily to a set of flags?

Author: simply that there are no developer specific logs in any logic outside of `--timeframe`, `--from` and `--to`

Author: reworded
